Ability to Queue Messages to different containers

shubham3121 opened the pull request:
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

shubham3121 committed:
extend consumer to connect to broker and specific queue

shubham3121 committed:
refactor Producer for multiple queues

shubham3121 committed:
add experimental notebooks

rasswanth-s committed:
added small zmq example

rasswanth-s committed:
added small zmq example

shubham3121 committed:
pass consumer service name to node and zmq client config
pass worker pool name as part of QueueItem
extract pool name from QueueItem and extract the corresponding worker
pass env vars to custom containers
pass network mode to custom containers

shubham3121 committed:
register service when worker is ready

shubham3121 committed:
simplify logic to process messages added to a service

shubham3121 committed:
remove unused code

shubham3121 committed:
Merge branch 'dev' into cw-multiple-queues

shubham3121 committed:
add pool to Queue as LinkedObject

shubham3121 committed:
add evaluate can run condition for job execution at consumer entry
- track worker uid at job level
- pass syft worker uid to consumer container as env var

shubham3121 committed:
Merge branch 'dev' into cw-multiple-queues

rasswanth-s commented on packages/syft/src/syft/node/node.py:
```diff
@@ -516,6 +526,7 @@ def named(
                     create_producer=create_producer,
                     queue_port=queue_port,
                     n_consumers=n_consumers,
+                    consumer_service=consumer_service,
```
NIT: should we also add consumer_service to also the init of Node Class.

shubham3121 replied:
Not required as `consumer_service` is part initialized as part of QueueConfig.

shubham3121 committed:
fix breaking ZMQ producer/consumer methods
- fix heartbeat timeout and expiry
- fix spinning workers when container not present
- set dev mode flag to custom containers

shubham3121 committed:
add method to create a default worker image using woker_cpu.dockerfile
- add method to create a default worker pool using default worker image
- implement stash method to get syft image by worker config

shubham3121 committed:
add READ ALL permission to syft worker pool and syft worker image objects

shubham3121 committed:
fix default pool creation
- ignore duplication error when creating admin details
- comment mounting steps in worker_cpu.dockerfile

shubham3121 committed:
added method to add consumer for a service at Node class
- pass queue port to consumer environment

shubham3121 committed:
only create default pool if no consumer service is defined
- add method to get queue address
- generalize add consumer method
- fix bug with worker pool not attaching to queueitem
- fix extracting message before passing to message handler
- create inmemory workers if backend container doesn't exists

shubham3121 committed:
pass default item to ActionQueueItem

shubham3121 committed:
orchestrate worker based on environment flag
- add in-memory worker flag to hagrid cli, node settings and Node class
- cleanup run_container_using_docker method and related utils
- hardcode workercpu dockerfile as string for default image

shubham3121 committed:
loop over n consumers for custom services

shubham3121 committed:
add woker stash
- update workerpool to track workers as LinkedObjects
- skip deleting containers if in memory workers
- skip logging containers if in memory workers
- cleanup worker service

shubham3121 committed:
pass syft worker uid to producer on W_READY
pass syft worker uid to handle message

shubham3121 committed:
integrate changes from PR https://github.com/OpenMined/PySyft/pull/8350

shubham3121 committed:
Merge branch 'dev' into cw-multiple-queues

shubham3121 committed:
fix missing worker stash argument to add_consumers method

shubham3121 committed:
fix context close during for ZMQConsumer and producer

shubham3121 committed:
skip raising value error if consumer count is zero

shubham3121 committed:
set unique identity to each Consumer/Producer object during socket creation

shubham3121 committed:
make send method consistent to use send_to_worker method in ZMQProducer
- fix zmq tests

shubham3121 committed:
- skip building image if inmemory workers

shubham3121 committed:
fix creation of queueitem object in queue stash tests

shubham3121 committed:
Merge branch 'dev' into cw-multiple-queues

shubham3121 committed:
calcuate syft version tag to the hardcoded dockerfile during image build
- skip build image if working with in memory workers

shubham3121 committed:
pass in memory worker flag to serve_node and run_uvircon method

shubham3121 committed:
fix container notebook in apis

shubham3121 committed:
remove assert statement from delete worker

shubham3121 committed:
move worker creation process for a pool to a method
- added a new api to add workers to existing worker pool

shubham3121 committed:
fix container images notebook
- fix returning wrong variable in _create_workers_in_pool

shubham3121 committed:
adjust container image notebook for stack container tests

shubham3121 committed:
Merge branch 'dev' into cw-multiple-queues

shubham3121 committed:
init docker client only when running container based workers

shubham3121 committed:
refactor start worker api to work with worker pool service

shubham3121 committed:
fix bug in add_worker API
